src/Compilers/CSharp/Test/Emit/PDB/PDBTupleTests.cs

```diff
@@ -262,5 +263,83 @@ static void F(System.Collections.Generic.IEnumerable<(int a, int b)> ie)
   </methods>
 </symbols>"));
         }
+
+        [Fact, WorkItem(17947, "https://github.com/dotnet/roslyn/issues/17947")]
+        public void VariablesAndConstantsInUnreachableCode()
+        {
+            string source = @"
+class C
+{
+    void F()
+    {
+        (int a, int b)[] v1 = null;
+        const (int a, int b)[] c1 = null;
+
+        throw null;
+
+        (int a, int b)[] v2 = null; 
+        const (int a, int b)[] c2 = null;
+
+        { 
+            (int a, int b)[] v3 = null; 
+            const (int a, int b)[] c3 = null;
+        }
+    }
+}
+";
+            var c = CreateStandardCompilation(source, new[] { ValueTupleRef, SystemRuntimeFacadeRef }, options: TestOptions.DebugDll);
+            var v = CompileAndVerify(c);
+            v.VerifyIL("C.F", @"
+{
+  // Code size        5 (0x5)
+  .maxstack  1
+  .locals init ((int a, int b)[] V_0, //v1
+                (int a, int b)[] V_1, //v2
```

Comment:
Yes, me too. @VSadov said we don't bother to remove variables defined in unreachable code.